index.js

```diff
@@ -207,12 +266,85 @@ class instance extends instance_skel {
 			}
 		}
 
+		let jsonResultDataHandler = (e, result) => {
+			if (e !== null) {
+				this.log('error', `HTTP ${action.action.toUpperCase()} Request failed (${e.message})`)
+				this.status(this.STATUS_ERROR, result.error.code)
+			} else {
+				this.status(this.STATUS_OK)
+
+				debug('processing JSON result.data')
+
+				// retrieve the variable name in which to store json result.data
+				this.system.emit('variable_parse', action.options.jsonResultDataVariable, (value) => {
+					jsonResultDataVariable = value
+				})
+
+				// store json result data into retrieved dedicated custom variable
+				if (jsonResultDataVariable !== '') {
+					debug('jsonResultDataVariable', jsonResultDataVariable)
+					let jsonResultData = JSON.stringify(result.data)
+					this.system.emit('custom_variable_set_value', jsonResultDataVariable, jsonResultData)
+				}
+			}
+		}		
+
 		let options = {
 			connection: {
-				rejectUnauthorized: self.config.rejectUnauthorized,
+				//TODO: verify this trivial fix (replaced 'self' with 'this')
+				rejectUnauthorized: this.config.rejectUnauthorized,
 			},
 		}
 
+// ================================================================================================
+		//TODO: this seems to be better implemented by core-team within internal module,
+		//      similar to "Set Custom Variable Value" etc.
+		//TODO: it looks like the custom-var handling should be moved to an own module,
+		//      like "custom-vars" or "vars".
+
+		// extract value from the stored json response data, assign to target variable		
+		//TODO: add error-handling
+		if (action.action === 'setvar') {
+
+			// get the variable(!) that holds the data
+			this.system.emit('variable_parse', action.options.jsonResultDataVariable, (value) => {
```

Comment:
I thought that this is the way to access variable values. https://github.com/bitfocus/companion/wiki/Variables has not much info, so I looked in other modules for reference. This looks a bit strange, but seemed to work fine.

How would I access the value of a variable in a simpler manner?

Reply:
Its the correct way to parse a string like `Value is: $(internal:custom_test)` replacing the variables with their values.
But that isnt what you want to do here. 

Lets say there is a customVariable called 'test' that you want to write to.
That means that `action.options.jsonResultDataVariable` has a value of 'test' because thats the variable you selected in the ui.
Then you do a `variable_parse` of that, and there are no variables to parse (the `$(internal:custom_test)` syntax) so it is guaranteed to return the input string back to you. But also, would you really want the value here? Lets say the value of the custom variable is `{}`, is that any use to you here?